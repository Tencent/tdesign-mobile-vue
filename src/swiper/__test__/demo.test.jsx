/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/swiper/demos/base.vue';
import cardVue from '@/swiper/demos/card.vue';
import controlVue from '@/swiper/demos/control.vue';
import customVue from '@/swiper/demos/custom.vue';
import fractionVue from '@/swiper/demos/fraction.vue';
import mobileVue from '@/swiper/demos/mobile.vue';
import outsideVue from '@/swiper/demos/outside.vue';
import verticalVue from '@/swiper/demos/vertical.vue';

const mapper = {
  baseVue,
  cardVue,
  controlVue,
  customVue,
  fractionVue,
  mobileVue,
  outsideVue,
  verticalVue,
};

describe('Swiper', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Swiper ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
