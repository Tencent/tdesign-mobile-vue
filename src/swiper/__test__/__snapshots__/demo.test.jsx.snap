// Vitest Snapshot v1

exports[`Swiper > Swiper baseVue demo works fine 1`] = `
<div
  style="padding: 0px 16px;"
>
  <div
    class="t-swiper t-swiper--default"
  >
    <div
      class="t-swiper__container"
      style="flex-direction: row; transition: none; height: auto;"
    >
      
      
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      
      
    </div>
    
    <!---->
    <span
      class="t-swiper-nav--horizontal t-swiper-nav__dots t-swiper-nav--bottom"
    >
      
      
      
      
      <!---->
    </span>
    
  </div>
</div>
`;

exports[`Swiper > Swiper cardVue demo works fine 1`] = `
<div
  class="t-swiper t-swiper--card"
>
  <div
    class="t-swiper__container"
    style="flex-direction: row; transition: none; height: auto;"
  >
    
    
    <div
      class="t-swiper-item"
      style=""
    >
      
      <img
        class="img"
        src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
      />
      
    </div>
    <div
      class="t-swiper-item"
      style=""
    >
      
      <img
        class="img"
        src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
      />
      
    </div>
    <div
      class="t-swiper-item"
      style=""
    >
      
      <img
        class="img"
        src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
      />
      
    </div>
    <div
      class="t-swiper-item"
      style=""
    >
      
      <img
        class="img"
        src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
      />
      
    </div>
    <div
      class="t-swiper-item"
      style=""
    >
      
      <img
        class="img"
        src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
      />
      
    </div>
    
    
  </div>
  
  <!---->
  <span
    class="t-swiper-nav--horizontal t-swiper-nav__dots t-swiper-nav--bottom"
  >
    
    
    
    
    <!---->
  </span>
  
</div>
`;

exports[`Swiper > Swiper controlVue demo works fine 1`] = `
<div
  style="padding: 0px 16px;"
>
  <div
    class="t-swiper t-swiper--default"
  >
    <div
      class="t-swiper__container"
      style="flex-direction: row; transition: none; height: auto;"
    >
      
      
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      
      
    </div>
    
    <span
      class="t-swiper-nav__btn"
    >
      <span
        class="t-swiper-nav__btn--prev"
      />
      <span
        class="t-swiper-nav__btn--next"
      />
    </span>
    <!---->
    
  </div>
</div>
`;

exports[`Swiper > Swiper customVue demo works fine 1`] = `
<div
  style="padding: 0px 16px;"
>
  <div
    class="t-swiper t-swiper--default"
  >
    <div
      class="t-swiper__container"
      style="flex-direction: row; transition: none; height: auto;"
    >
      
      
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      
      
    </div>
    
    <!---->
    <span
      class="t-swiper-nav--horizontal t-swiper-nav__dots-bar t-swiper-nav--bottom"
    >
      
      
      
      
      <!---->
    </span>
    
  </div>
</div>
`;

exports[`Swiper > Swiper fractionVue demo works fine 1`] = `
<div
  style="padding: 0px 16px;"
>
  <div
    class="t-swiper t-swiper--default"
  >
    <div
      class="t-swiper__container"
      style="flex-direction: row; transition: none; height: auto;"
    >
      
      
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      
      
    </div>
    
    <!---->
    <span
      class="t-swiper-nav--horizontal t-swiper-nav__fraction t-swiper-nav--bottom-right"
    >
      <!---->
      <span>
        1/0
      </span>
    </span>
    
  </div>
</div>
`;

exports[`Swiper > Swiper mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Swiper 轮播
  </h1>
  <p
    class="summary"
  >
    用于循环轮播一组图片或内容，也可以滑动进行切换，轮播动效时间可以设置。
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        点状（dots）
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        style="padding: 0px 16px;"
      >
        <div
          class="t-swiper t-swiper--default"
        >
          <div
            class="t-swiper__container"
            style="flex-direction: row; transition: none; height: auto;"
          >
            
            
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            
            
          </div>
          
          <!---->
          <span
            class="t-swiper-nav--horizontal t-swiper-nav__dots t-swiper-nav--bottom"
          >
            
            
            
            
            <!---->
          </span>
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        条状（dots-bar）
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        style="padding: 0px 16px;"
      >
        <div
          class="t-swiper t-swiper--default"
        >
          <div
            class="t-swiper__container"
            style="flex-direction: row; transition: none; height: auto;"
          >
            
            
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            
            
          </div>
          
          <!---->
          <span
            class="t-swiper-nav--horizontal t-swiper-nav__dots-bar t-swiper-nav--bottom"
          >
            
            
            
            
            <!---->
          </span>
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        分式（fraction）
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        style="padding: 0px 16px;"
      >
        <div
          class="t-swiper t-swiper--default"
        >
          <div
            class="t-swiper__container"
            style="flex-direction: row; transition: none; height: auto;"
          >
            
            
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            
            
          </div>
          
          <!---->
          <span
            class="t-swiper-nav--horizontal t-swiper-nav__fraction t-swiper-nav--bottom-right"
          >
            <!---->
            <span>
              1/0
            </span>
          </span>
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        切换按钮（controls）
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        style="padding: 0px 16px;"
      >
        <div
          class="t-swiper t-swiper--default"
        >
          <div
            class="t-swiper__container"
            style="flex-direction: row; transition: none; height: auto;"
          >
            
            
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            
            
          </div>
          
          <span
            class="t-swiper-nav__btn"
          >
            <span
              class="t-swiper-nav__btn--prev"
            />
            <span
              class="t-swiper-nav__btn--next"
            />
          </span>
          <!---->
          
        </div>
      </div>
      
    </div>
  </div>
  <!-- &lt;tdesign-demo-block summary="卡片模式"&gt;
      &lt;card /&gt;
    &lt;/tdesign-demo-block&gt; -->
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件样式
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        垂直模式
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        style="padding: 0px 16px;"
      >
        <div
          class="t-swiper t-swiper--default"
        >
          <div
            class="t-swiper__container"
            style="flex-direction: column; transition: none; height: auto;"
          >
            
            
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style=""
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            
            
          </div>
          
          <!---->
          <span
            class="t-swiper-nav--vertical t-swiper-nav__dots-bar t-swiper-nav--right"
          >
            
            
            
            
            <!---->
          </span>
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        outside模式
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        style="padding: 0px 16px;"
      >
        <div
          class="t-swiper t-swiper--default t-swiper--outside"
        >
          <div
            class="t-swiper__container"
            style="flex-direction: row; transition: none; height: auto;"
          >
            
            
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
              />
              
            </div>
            <div
              class="t-swiper-item"
              style="height: 192px;"
            >
              
              <img
                class="img"
                src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
              />
              
            </div>
            
            
          </div>
          
          <!---->
          <span
            class="t-swiper-nav--horizontal t-swiper-nav__dots t-swiper-nav--bottom t-swiper-nav--outside"
          >
            
            
            
            
            <!---->
          </span>
          
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Swiper > Swiper outsideVue demo works fine 1`] = `
<div
  style="padding: 0px 16px;"
>
  <div
    class="t-swiper t-swiper--default t-swiper--outside"
  >
    <div
      class="t-swiper__container"
      style="flex-direction: row; transition: none; height: auto;"
    >
      
      
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style="height: 192px;"
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      
      
    </div>
    
    <!---->
    <span
      class="t-swiper-nav--horizontal t-swiper-nav__dots t-swiper-nav--bottom t-swiper-nav--outside"
    >
      
      
      
      
      <!---->
    </span>
    
  </div>
</div>
`;

exports[`Swiper > Swiper verticalVue demo works fine 1`] = `
<div
  style="padding: 0px 16px;"
>
  <div
    class="t-swiper t-swiper--default"
  >
    <div
      class="t-swiper__container"
      style="flex-direction: column; transition: none; height: auto;"
    >
      
      
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper2.png"
        />
        
      </div>
      <div
        class="t-swiper-item"
        style=""
      >
        
        <img
          class="img"
          src="https://tdesign.gtimg.com/mobile/demos/swiper1.png"
        />
        
      </div>
      
      
    </div>
    
    <!---->
    <span
      class="t-swiper-nav--vertical t-swiper-nav__dots-bar t-swiper-nav--right"
    >
      
      
      
      
      <!---->
    </span>
    
  </div>
</div>
`;
